src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -235,6 +236,9 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
 
         auto pass_config = manager.get_pass_config();
         pass_config->disable<ov::pass::EyeDecomposition>();
+
+        // in order to keep ShapeOf precision sensitive subgraphs in FP32
+        pass_config->enable<ov::pass::MarkPrecisionSensitiveSubgraphs>();
         pass_config->enable<ov::pass::ConvertCompressedOnlyToLegacy>();
```

Comment:
Not good way to enable transformation here that should be registered inside another transformation.
May be it is better to configure `ConvertCompressedOnlyToLegacy` by adding new arguments for it and to configure it from outside

Reply:
Agreed and addressed in the new PR https://github.com/openvinotoolkit/openvino/pull/14822